packages/amplify-category-function/src/provider-utils/awscloudformation/utils/layerParams.ts

```diff
@@ -15,3 +18,165 @@ export enum Permissions {
+}
+
+export interface LayerVersionMetadata {
+  permissions: LayerPermission[];
+  listAccoutAccess: () => string[];
+  listOrgAccess: () => string[];
+  isPrivate: () => boolean;
+  isPublic: () => boolean;
+}
+
+export type LayerPermission = PrivateLayer | PublicLayer | AccountsLayer | OrgsLayer;
+
+export interface PrivateLayer {
+  type: Permissions.private;
+}
+
+export interface PublicLayer {
+  type: Permissions.public;
+}
+
+export interface AccountsLayer {
+  type: Permissions.awsAccounts;
+  accounts: string[];
+}
+
+export interface OrgsLayer {
+  type: Permissions.awsOrg;
+  orgs: string[];
+}
+
+class LayerState implements LayerMetadata {
+  versionMap: Map<number, LayerVersionMetadata> = new Map();
+  runtimes: FunctionRuntime[];
+  layerMetaData: LayerVersionMetadata;
+  constructor(obj) {
+    this.runtimes = obj.runtimes;
+    Object.entries(obj.layerVersionsMap).forEach(([key, value]) => {
+      this.layerMetaData = new LayerVersionState(value);
+      this.versionMap.set(Number(key), this.layerMetaData);
+    });
+  }
+
+  getVersion(version: number): LayerVersionMetadata {
+    return this.versionMap.get(version);
+  }
+
+  listVersions(): number[] {
+    return Array.from(this.versionMap.keys());
+  }
+  getLatestVersion(): number {
+    return Array.from(this.versionMap.keys()).reduce((a, b) => Math.max(a, b));
+  }
+}
+
+class LayerVersionState implements LayerVersionMetadata {
+  permissions: LayerPermission[];
+  constructor(value) {
+    this.permissions = [];
+    value.forEach(permission => {
+      if (permission.type === Permissions.public) {
+        const permissionPublic: PublicLayer = {
+          type: Permissions.public,
+        };
+        this.permissions.push(permissionPublic);
+      }
+      if (permission.type === Permissions.private) {
+        const permissionPrivate: PrivateLayer = {
+          type: Permissions.private,
+        };
+
+        this.permissions.push(permissionPrivate);
+      }
+      if (permission.type === Permissions.awsOrg) {
+        const orgsPermissions: OrgsLayer = {
+          type: Permissions.awsOrg,
+          orgs: permission.orgs,
+        };
+        this.permissions.push(orgsPermissions);
+      }
+      if (permission.type === Permissions.awsAccounts) {
+        const accountsPermissions: AccountsLayer = {
+          type: Permissions.awsAccounts,
+          accounts: permission.accounts,
+        };
+        this.permissions.push(accountsPermissions);
+      }
+    });
+  }
+
+  listAccoutAccess(): string[] {
+    let accounts: string[] = [];
+    if (!(this.permissions === undefined || this.permissions.length == 0)) {
+      this.permissions.forEach(permission => {
+        if (permission.type === Permissions.awsAccounts) {
+          accounts = permission.accounts;
+        }
+      });
+    }
+    return accounts;
+  }
+
+  listOrgAccess(): string[] {
+    let orgs: string[] = [];
+    if (!(this.permissions === undefined || this.permissions.length == 0)) {
```

Comment:
`!(this.permissions === undefined || this.permissions.length == 0)` uses == instead of ===.

You can simplify this to:
`this.permissions && this.permissions.length > 0`

Reply:
as mentioned above, you don't need this check at all